lib/logstash/gemfile.rb

```diff
@@ -0,0 +1,173 @@
+module LogStash
+
+  class GemfileError < StandardError; end
+
+  class Gemfile
+    attr_accessor :gemset
+
+    HEADER = \
+      "# This is a Logstash generated Gemfile.\n" + \
+      "# If you modify this file manually all comments and formatting will be lost.\n\n"
+
+    # @params io [IO] any IO object that supports read, write, truncate, rewind
+    def initialize(io)
+      @io = io
+      @gemset = nil
+    end
+
+    def load
+      @gemset ||= Dsl.parse(@io.read)
```

Comment:
Even if I find this `Dsl` ugly, I think the correct naming rule would be `Dsl`.. But who I am to judge that ;)
as a sidenote, one of the library we are using when that route, the `aws-sdk` was using `AWS` in v1 and in v2 they are using `Aws`.